File: platforms/kotlin/examples/core/pass/set_viewport.kt

import org.fragmentcolor.*

val renderer = Renderer()
val target = renderer.createTextureTarget(64u, 64u)

val shader = Shader.default()
val pass = Pass("clipped")
pass.addShader(shader)

pass.setViewport(ScreenRegion(minX=0u, minY=0u, maxX=32u, maxY=32u))

renderer.render(pass, target)

File: platforms/swift/examples/core/pass/set_viewport.swift

import FragmentColor

let renderer = Renderer()
let target = try await renderer.createTextureTarget([64, 64])

let shader = Shader.default()
let pass = Pass("clipped")
pass.addShader(shader)

pass.setViewport([(0, 0), (32, 32)])

renderer.render(pass, target)